packages/Button/src/Button.tsx

```diff
@@ -333,85 +320,81 @@ function usesLinkElement(
-  merging them together with ButtonProps (see LibraryManagedAttributes and MergePropTypes in @types/react).
-  Unfortunately, this merging uses keyof, which appears to drop the [key: string] index signature, meaning TS won't
-  allow us to pass unrecognized props down to custom components when we're using the `as` prop. This workaround avoids
-  the attempt at merging, while still getting runtime type-checking for non-TS consumers of the library.
-  */
-  static propTypes: any = {
-    variant: PropTypes.oneOf(['default', 'primary', 'info', 'danger', 'dark']),
-    size: PropTypes.oneOf(['xsmall', 'small', 'normal', 'large']),
-    className: PropTypes.string,
-    children: PropTypes.node,
-    disabled: PropTypes.bool,
-    as: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
-    href: PropTypes.string,
+export default function Button(props: ButtonProps) {
+  const {
+    className = '',
+    children = null,
+    disabled = false,
+    variant = Variant.Default,
+    size = Size.Normal,
+  } = props;
+
+  const commonProps = {
+    className: cx(
+      baseStyle,
+      buttonSizes[size],
+      buttonVariants[variant],
+      className,
+    ),
+    disabled,
+    'aria-disabled': disabled,
   };
 
-  render() {
-    const { className, children, disabled, variant, size } = this.props;
-
-    const commonProps = {
-      className: cx(
-        baseStyle,
-        buttonSizes[size],
-        buttonVariants[variant],
-        className,
-      ),
-      disabled,
-      'aria-disabled': disabled,
-    };
-
-    const rest = omit(this.props, [
-      'as',
-      'className',
-      'disabled',
-      'size',
-      'variant',
-      'children',
-    ]);
-
-    if (usesCustomElement(this.props)) {
-      const Root = this.props.as;
+  const rest = omit(props, [
+    'as',
+    'className',
+    'disabled',
+    'size',
+    'variant',
+    'children',
+  ]);
 
-      return (
-        <Root {...rest} {...commonProps}>
-          {children}
-        </Root>
-      );
-    }
+  if (usesCustomElement(props)) {
+    const Root = props.as;
 
-    if (usesLinkElement(this.props)) {
-      return (
-        <a {...rest as HTMLElementProps<'a'>} {...commonProps}>
-          {children}
-        </a>
-      );
-    }
+    return (
+      <Root {...rest} {...commonProps}>
+        {children}
+      </Root>
+    );
+  }
 
-    // NOTE(JeT): The button's `type` will be overridden if it is in the passed-in props
+  if (usesLinkElement(props)) {
     return (
-      <button
-        type="button"
-        {...rest as HTMLElementProps<'button'>}
-        {...commonProps}
-      >
+      <a {...(rest as HTMLElementProps<'a'>)} {...commonProps}>
         {children}
-      </button>
+      </a>
     );
   }
+
+  // NOTE(JeT): The button's `type` will be overridden if it is in the passed-in props
+  return (
+    <button
+      type="button"
+      {...(rest as HTMLElementProps<'button'>)}
```

Comment:
Does `HTMLElementProps` include event handler props like `onClick`?

Reply:
@jetpacmonkey

Reply:
It should, yes. For this specific line it's just a type-cast that doesn't actually affect anything, but `HTMLElementProps<'button'>` is equivalent to `React.ButtonHTMLAttributes<HTMLButtonElement>` (it's pulling it out of this definition [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/fa812c285d13280dc12c285d4538f209bbe1e12b/types/react/index.d.ts#L2514))

Reply:
Just double-checking 👍